LandSandBoat/server
# [cpp/sql] Spellreq setting for main-job-only spells

**_I affirm:_**

## What does this pull request do?

Rather than a lua restriction for the issue found [here](https://github.com/LandSandBoat/server/pull/5750), this PR adds a new spellreq setting (not a flag) to define spells only available to main job

## Steps to test these changes

import sql and compile PR, see that `/ma geo-regen`, etc doesn't work. It doesn't show up in the menu due to the DATs, but was still possible to cast via macro

![image](https://github.com/LandSandBoat/server/assets/131182600/c544c8e8-25f3-496e-84c8-21efc00be2cc)
